### PR TITLE
Topic model

### PR DESCRIPTION
**Pull request by charleslparker**
Bringing everything in line with the changes to the node bindings.  The main thing was just making the test use the unicode characters.  The rest is just a wee refactoring so the code in both places looks as identical as possible.

### CODE REVIEW

File: bigml/topicmodel.py [hunk 1]
```diff
@@ -334,30 +348,23 @@ def infer(self, list_of_indices):
             updates = min(MAX_UPDATES, max(MIN_UPDATES, updates));
```
Comment (mmerce) [inline bigml/topicmodel.py]: ending with semicolon

File: bigml/topicmodel.py [hunk 2]
```diff
@@ -334,30 +348,23 @@ def infer(self, list_of_indices):
...
+        uniform_counts = self.sample_uniform(doc, updates, rng)
 
         # Burn-in
-        for _ in range(updates):
-            for term in doc:
-                topic = self.sample_topic(term, uniform_counts, normalizer, rng)
-                burn_counts[topic] += 1
-
+        burn_counts = self.sample_topics(doc,
+                                        uniform_counts,
```
Comment (mmerce) [inline bigml/topicmodel.py]: continuation lines are not well indented

File: bigml/topicmodel.py [hunk 3]
```diff
@@ -334,30 +348,23 @@ def infer(self, list_of_indices):
...
+                                        normalizer,
+                                        updates,
+                                        rng)
         # Sampling
-        for _ in range(updates):
-            for term in doc:
-                topic = self.sample_topic(term, burn_counts, normalizer, rng)
-                sample_counts[topic] += 1
+        sample_counts = self.sample_topics(doc,
+                                          burn_counts,
```
Comment (mmerce) [inline bigml/topicmodel.py]: continuation lines are not well indented

**Comment by charleslparker** [comment]
Nits picked - thanks for spotting the indentation miss!

**Pull request merged**